File: JSPM.Web/Modules/Syllabus/Topics/TopicsForm.cs
using Serenity.ComponentModel;

namespace JSPM.Syllabus.Forms;

[FormScript("Syllabus.Topics")]
[BasedOnRow(typeof(TopicsRow), CheckNames = true)]
public class TopicsForm
{
    public int SubjectId { get; set; }
    public int ChapterId { get; set; }
    public string TopicName { get; set; }
    public string Description { get; set; }
    public int Index { get; set; }
}

File: JSPM.Web/Modules/ServerTypes/Syllabus/TopicsForm.ts
import { IntegerEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface TopicsForm {
    SubjectId: IntegerEditor;
    ChapterId: IntegerEditor;
    TopicName: StringEditor;
    Description: StringEditor;
    Index: IntegerEditor;
}

export class TopicsForm extends PrefixedContext {
    static formKey = 'Syllabus.Topics';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TopicsForm.init)  {
            TopicsForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;

            initFormType(TopicsForm, [
                'SubjectId', w0,
                'ChapterId', w0,
                'TopicName', w1,
                'Description', w1,
                'Index', w0
            ]);
        }
    }
}